src/mdtest.c

```diff
@@ -1210,129 +1238,141 @@ void summarize_results(int iterations) {
-                        if (max < curr) {
-                            max =  curr;
-                        }
-                        sum += curr;
-                    }
-                }
-                mean = sum / (iterations * size);
-                for (k=0; k<size; k++) {
-                    for (j = 0; j < iterations; j++) {
-                        var += pow((mean -  all[(k*tableSize*iterations)
-                                                + (j*tableSize) + i]), 2);
-                    }
-                }
-                var = var / (iterations * size);
-                sd = sqrt(var);
-                switch (i) {
-                case 0: strcpy(access, "Directory creation:"); break;
-                case 1: strcpy(access, "Directory stat    :"); break;
-                    /* case 2: strcpy(access, "Directory read    :"); break; */
-                case 2: ;                                      break; /* N/A */
-                case 3: strcpy(access, "Directory removal :"); break;
-                case 4: strcpy(access, "File creation     :"); break;
-                case 5: strcpy(access, "File stat         :"); break;
-                case 6: strcpy(access, "File read         :"); break;
-                case 7: strcpy(access, "File removal      :"); break;
-                default: strcpy(access, "ERR");                 break;
-                }
-                if (i != 2) {
-                    fprintf(out_logfile, "   %s ", access);
-                    fprintf(out_logfile, "%14.3f ", max);
-                    fprintf(out_logfile, "%14.3f ", min);
-                    fprintf(out_logfile, "%14.3f ", mean);
-                    fprintf(out_logfile, "%14.3f\n", sd);
-                    fflush(out_logfile);
-                }
-                sum = var = 0;
+    /* if directories only access, skip entries 4-7 (the file tests) */
+    if (dirs_only && !files_only) {
+        stop = 4;
+    } else {
+        stop = 8;
+    }
 
-        }
+    /* special case: if no directory or file tests, skip all */
+    if (!dirs_only && !files_only) {
+        start = stop = 0;
+    }
 
-        /* calculate tree create/remove rates */
-        for (i = 8; i < tableSize; i++) {
+    for (i = start; i < stop; i++) {
             min = max = all[i];
-            for (j = 0; j < iterations; j++) {
-                if(print_time){
-                  curr = summary_table[j].time[i];
-                }else{
-                  curr = summary_table[j].rate[i];
-                }
-
-                if (min > curr) {
-                    min = curr;
-                }
-                if (max < curr) {
-                    max =  curr;
+            for (k=0; k < size; k++) {
+                for (j = 0; j < iterations; j++) {
+                    curr = all[(k*tableSize*iterations)
+                               + (j*tableSize) + i];
+                    if (min > curr) {
+                        min = curr;
+                    }
```

Comment:
Similarly, I wonder if these if statements with just a single op should be one-liners?  Just makes code more condense so easier to scan it quickly.